blockchain/vm/evm_test.go

```diff
@@ -97,3 +56,79 @@ func TestPrecompiledContractAddressMapping(t *testing.T) {
 		assert.Equal(t, tc.expectErr, err)
 	}
 }
+
+func TestPrecompiledContractAddressMapping(t *testing.T) {
+	var (
+		// Test Input
+		vmLogInput    = []byte("Hello")
+		feePayerInput = []byte("")
+		validateInput = common.Hex2Bytes("00000000000000000000000000000001234567890000000000000000000000000000000000000000000000000000000000000000db38e383310a432f1cec0bfbd765fe67c9932b02295b1f2062a785eec288b2f962d578d163a8b13c9cee127e0b24c01d1906979ac3bc6a763bf95c50999ca97c01")
+		// Expected Output
+		vmLogOutput    = ""
+		feePayerOutput = "000000000000000000000000636f6e7472616374"
+		validateOutput = "00"
+		// Test ChainConfig
+		config = &params.ChainConfig{IstanbulCompatibleBlock: Block5} // Set IstanbulCompatible block number as '5'
+	)
+
+	runPrecompiledContractTestWithHFCondition(t, config, []TestData{
+		// Expect Normal Behavior
+		// Condition 1. Caller Contract Deploy - before IstanbulCompatible, Call - before istanbulCompatible
+		{"0x009", vmLogInput, false, Block4, 200, vmLogOutput, nil},
+		{"0x00a", feePayerInput, false, Block4, 300, feePayerOutput, nil},
+		{"0x00b", validateInput, false, Block4, 5000, validateOutput, nil},
+		// Condition 2. Caller Contract Deploy - before IstanbulCompatible, Call - after istanbulCompatible
+		{"0x009", vmLogInput, false, Block5, 200, vmLogOutput, nil},
+		{"0x00a", feePayerInput, false, Block5, 300, feePayerOutput, nil},
+		{"0x00b", validateInput, false, Block5, 5000, validateOutput, nil},
+		// Condition 3. Caller Contract Deploy - after IstanbulCompatible, Call - after istanbulCompatible
+		{"0x3fd", vmLogInput, true, Block5, 200, vmLogOutput, nil},
+		{"0x3fe", feePayerInput, true, Block5, 300, feePayerOutput, nil},
+		{"0x3ff", validateInput, true, Block5, 5000, validateOutput, nil},
+
+		// Expect Error Behavior
+		// Condition 1. Caller Contract Deploy - before IstanbulCompatible, Call - before istanbulCompatible
+		{"0x3fd", vmLogInput, false, Block4, 0, "", kerrors.ErrPrecompiledContractAddress},
+		{"0x3fe", feePayerInput, false, Block4, 0, "", kerrors.ErrPrecompiledContractAddress},
+		{"0x3ff", validateInput, false, Block4, 0, "", kerrors.ErrPrecompiledContractAddress},
+		// Condition 2. Caller Contract Deploy - before IstanbulCompatible, Call - after istanbulCompatible
+		{"0x3fd", vmLogInput, false, Block5, 0, "", kerrors.ErrPrecompiledContractAddress},
+		{"0x3fe", feePayerInput, false, Block5, 0, "", kerrors.ErrPrecompiledContractAddress},
+		{"0x3ff", validateInput, false, Block5, 0, "", kerrors.ErrPrecompiledContractAddress},
+		// Condition 3. Caller Contract Deploy - after IstanbulCompatible, Call - after istanbulCompatible
+		{"0x009", vmLogInput, true, Block5, math.MaxUint64, "", errors.New("invalid input length")},
```

Comment:
@yoomee1313 Why does this test case have `math.MaxUint64` gas?

Reply:
@KimKyungup `invalid input length` error occurs when executing blake2f precompiled contract. That is, it is a VM error, and if any vm error occurs, all remaining gas is consumed. In this tc, gas limit is Uint64, so it consumes MaxUint64.

> // When an error was returned by the EVM or when setting the creation code
> // above we revert to the snapshot and consume any gas remaining. Additionally
> // when we're in homestead this also counts for code storage gas errors.